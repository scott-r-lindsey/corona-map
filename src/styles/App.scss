
@import './main-panel.scss';
@import './left-sidebar.scss';

@import './breakpoints.scss';
@import './colors.scss';
@import './typography.scss';

* {
    box-sizing: border-box;
}
html, body, #root, .MuiGrid-root{
    height: 100%;
}



