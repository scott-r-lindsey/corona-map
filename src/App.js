import React, { useState, useEffect } from 'react';
import './styles/App.scss';
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import Main from './components/index.js';
import { home } from './lib/config.js';
import { primary, secondary} from './lib/colors.js';
import TrackedRoute from './components/TrackedRoute.js';

import {
  Redirect,
  BrowserRouter as Router,
  Switch,
} from "react-router-dom";

const fetch = require('node-fetch');
const dataUrl = '/data/latestfull.json';

const theme = createMuiTheme({
  palette: {
    primary: { main: primary },
    secondary: { main: secondary }
  },
  status: {
    danger: 'orange',
  },
});

function App() {


  const [data, setData] = useState(null);

  useEffect(() => {
    (async () => {
      await fetch(dataUrl, {method: 'GET'})
        .then(res => res.json())
        .then((data) => { setData(data, []) })
    })();
  }, []);

  return (
    <>
      { data ?
        <ThemeProvider theme={theme}>
          <Router>
            <Switch>
              <TrackedRoute path="/COVID-US/:when/:axis/:location">
                <Main data={data}/>
              </TrackedRoute>
              <TrackedRoute exact path="/">
                <Redirect to={home}/>
              </TrackedRoute>
            </Switch>
          </Router>
        </ThemeProvider> :
        <div>Loading...</div>
      }
    </>
  );
}

export default App;
