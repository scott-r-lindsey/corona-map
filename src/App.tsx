import React, { useState, useEffect } from 'react';
import './styles/App.scss';
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import { home } from './lib/config.js';
import { primary, secondary} from './lib/colors';
import TrackedRoute from './components/TrackedRoute';
import RouteValidator from './components/RouteValidator';
import { embellishData } from './lib/getMapValue';
import Veil from './components/Veil';
import VeilContext from './context/Veil';
import {
  Redirect,
  BrowserRouter as Router,
  Switch,
} from "react-router-dom";

const fetch = require('node-fetch');
const dataUrl = '/data/full.json';

const breakpoints = {
  xs: 0,
  sm: 680,
  md: 1060,
  lg: 1280,
  xl: 1920,
};

const theme = createMuiTheme({
  breakpoints: { values: breakpoints },
  palette: {
    primary: { main: primary },
    secondary: { main: secondary }
  },
});


function App(props) {

  const [data, setData] = useState(null);
  const [veil, setVeil] = useState(false);

  useEffect(() => {
    (async () => {

      const response = await fetch(dataUrl, {method: 'GET'});
      const result = await response.json();
      embellishData(result.state);
      embellishData(result.county);
      setData(result)

    })();
  }, []);

  return (
    <>
      { data ?
        <ThemeProvider theme={theme}>
          <VeilContext.Provider value={{veil, setVeil}}>
            <Router>
              <Switch>
                <TrackedRoute path="/:mode/:when/:axis/:quant/:location">
                  <RouteValidator data={data}/>
                </TrackedRoute>
                <TrackedRoute path="/">
                  <Redirect to={home}/>
                </TrackedRoute>
              </Switch>
            </Router>
            <Veil/>
          </VeilContext.Provider>
        </ThemeProvider> :
        <div style={{
          backgroundColor: primary,
          position:'fixed',
          top: 0,
          left: 0,
          right: 0,
          bottom: 0,
          display: 'flex',
          justifyContent: 'center',
        }}>
          <div className="loading-animation"><div></div><div></div><div></div></div>
        </div>
      }
    </>
  );
}

export default App;
