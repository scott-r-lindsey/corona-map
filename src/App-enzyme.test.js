import React from 'react';
import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router'
import { Route } from 'react-router-dom';
import data from '../fixture/full.json';
const { act } = require("react-dom/test-utils");
import { createMount } from '@material-ui/core/test-utils';
import {
  createMuiTheme,
  ThemeProvider as MuiThemeProvider
} from '@material-ui/core/styles';
import mediaQuery from 'css-mediaquery';

// -- load fetch mock before App ------------------------------------------
jest.mock('node-fetch', () => require('fetch-mock-jest').sandbox())
const fetchMock = require('node-fetch')
import App from './App';
// ------------------------------------------------------------------------

// -- mock mediaquery result ----------------------------------------------
function createMatchMedia(width) {
  return query => ({
    matches: mediaQuery.match(query, { width }),
    addListener: () => {},
    removeListener: () => {},
  });
}
// ------------------------------------------------------------------------

enzyme.configure({ adapter: new Adapter() });
const url = '/COVID-US/now/confirmed/total/united%20states';

describe ('App.js', () => {

  const dataUrl = '/data/full.json';
  let mount;

  beforeAll(() => {
    window.matchMedia = createMatchMedia(1600);
  });

  beforeEach(() => {
    mount = createMount();
    fetchMock.getOnce(dataUrl, data);
  })

  afterEach(() => {
    mount.cleanUp();
    fetchMock.reset();
    fetchMock.restore();
  });

  it('Loads the app', async () => {

    let wrapper;
    await act(async() => {

      wrapper = mount(
        <MemoryRouter initialEntries={[url]}>
          <App />
        </MemoryRouter>
      );
    });

    expect(fetchMock).toHaveFetched(dataUrl);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('div.loading-animation').html()).toBeTruthy();

  });
});
