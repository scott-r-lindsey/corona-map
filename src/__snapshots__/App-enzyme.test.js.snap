// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js Loads the app 1`] = `"<div style=\\"background-color: rgb(66, 66, 66); position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; display: flex; justify-content: center;\\"><div class=\\"loading-animation\\"><div></div><div></div><div></div></div></div>"`;
